torchgen/model.py

```diff
@@ -1176,6 +1375,43 @@ def strip_ret_annotation(r: Return) -> Return:
         if strip_view_copy_name and base_name.endswith("_copy"):
             base_name = base_name.replace("_copy", "")
 
+        # find mutable inputs that are not originally returned, and convert them to returns
+        returns_from_mutable_inputs = tuple(
+            # When we're grouping functions we strip the return names,
+            # but when we're generating the actual functional variants then we follow
+            # a convention for what to name the returns
+            Return(
+                name=f"{a.name}_out" if keep_return_names else None,
+                type=a.type,
+                annotation=None,
+            )
+            for a in itertools.chain(
+                # Order is important here (otherwise e.g. inplace with mutable args
+                # and out= with mutable args won't have the same signature)
+                [self.arguments.self_arg.argument]
+                if self.arguments.self_arg is not None
+                else [],
+                self.arguments.out,
+                self.arguments.post_self_positional,
+            )
+            if a.annotation is not None
+            and a.annotation.is_write
+            and not any(a.annotation == r.annotation for r in self.returns)
+        )
+        original_returns = tuple(map(strip_ret_annotation, self.returns))
+        # Ordering is important here. We expect the "mutable input" returns to come last.
+        returns = original_returns + returns_from_mutable_inputs
+
+        args_sig = self.arguments.signature(strip_default=strip_default)
+        # See Note [arange.start_step schema]
+        if str(self.name) == "arange.start_step":
+            args_sig = Arguments.parse(
+                str(args_sig).replace("Scalar step", "Scalar step=1")
+            )
+        # See Note [bernoulli.p schema]
+        if str(self.name) == "bernoulli.p":
+            args_sig = Arguments.parse(str(args_sig).replace("float p", "float p=0.5"))
```

Comment:
Both of the above two ops don't get grouped properly with their mutable variants. We could either:

(1) Just not provide functionalization/grouping support for them for now (not ideal because `arange` is pretty common and used, and also with these two fixes I was able to update LTC to only care about functional ops)
(2) Modify their schemas/C++ API names so they do get grouped properly (BC-breaking)
(3) special case their grouping here (solves the above two problems. It feels relatively harmless, and we can remove it if/when we're ok with making the BC change above)